JAICore/jaicore-ml/src/jaicore/ml/cache/ReproducibleInstances.java

```diff
@@ -0,0 +1,175 @@
+package jaicore.ml.cache;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import jaicore.ml.WekaUtil;
+import jaicore.ml.openml.OpenMLHelper;
+import weka.core.Instances;
+
+/**
+ * New Instances class to track splits and data origin.
+ * 
+ * @author jnowack
+ * @author mirko
+ * @author jonas
+ *
+ */
+public class ReproducibleInstances extends Instances {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 4318807226111536282L;
+	private List<Instruction> history = new LinkedList<>();
+	private boolean cacheStorage = true;
+	private boolean cacheLookup = true;
+	
+	private ReproducibleInstances(Instances dataset) {
+		super(dataset);
+	}
+	
+	/**
+	 * Creates a {@link ReproducibleInstances} Object based on the given History. Instructions that no not modify the Instances will be ignored (No evaluation will be done). 
+	 * 
+	 * @param history - List of INstructions used to create the original Instances
```

Comment:
why are these dashes everywhere in the javadoc of the methods? This is not part of the Oracle Javadoc conventions afaik.